week-1-ts-ecosystem/w01e03-object-oriented-structure/src/cart/Cart.ts

```diff
@@ -0,0 +1,49 @@
+import {ProductType, updateProductType} from "../types/product";
+import Product from "../product/Product";
+
+export enum TypeCart{
+  AUCTION="auction",
+  BUY_NOW="buy now",
+  FREE="free"
+}
```

Comment:
`enum` jest tutaj o tyle zasadny, że korzystasz z niego potem jako wartość. 
Jednak gdybyś przeprawił to na użycie klas i dziedziczenia w `Product`

Tak aby np. był bazowy `Product` a potem dziedziczenie: ten co się go da kupić z aukcji, kupić teraz i ten za free.
To wtedy Sama klasa determinowałaby typ produktu i nie potrzeba by było wprowadzać `enuma`